quickwit-indexing/src/source/mod.rs

```diff
@@ -216,7 +222,12 @@ impl Handler<Loop> for SourceActor {
     type Reply = ();
 
     async fn handle(&mut self, _message: Loop, ctx: &SourceContext) -> Result<(), ActorExitStatus> {
-        self.source.emit_batches(&self.batch_sink, ctx).await?;
+        let wait_for_opt = self.source.emit_batches(&self.batch_sink, ctx).await?;
+        if let Some(wait_for) = wait_for_opt {
+            // wait_for time can be more than a HEARTBEAT timeout.
+            ctx.protect_zone();
+            tokio::time::sleep(wait_for).await;
```

Comment:
could we use the self_schedule message instead? (as a replacement to  the send_self_message).

There are several benefit:
- simulating time in tests
- be able to send commands to the source
- it will help us profiling as the actor will be properly idle when it waits.

Reply:
Also can we return `Duration` and rely on `Duration` being 0 instead of `None`.